sns/src/main/scala/akka/stream/alpakka/sns/SnsPublishSinkStage.scala

```diff
@@ -0,0 +1,77 @@
+package akka.stream.alpakka.sns
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sns.AmazonSNSAsync
+import com.amazonaws.services.sns.model.{PublishRequest, PublishResult}
+
+import scala.concurrent.{Future, Promise}
+
+final class SnsPublishSinkStage(topicArn: String, snsClient: AmazonSNSAsync)
```

Comment:
This means we could dlete the sink stage and express it by using the flow in the factory method instead:
```
val flow: Flow[String, PublishResult, NotUsed] 
val sink: Sink[String, Future[Done]] = flow.toMat(Sink.ignore)(Keep.right)
```